frontend/src/components/Dish/DishForm.js

```diff
@@ -0,0 +1,136 @@
+import React, { useState, useContext } from "react";
+import userContext from "../../contexts/context";
+const _ = require("lodash");
+
+const apiUtil = require("../../util/apiUtil");
+function DishForm() {
+    const [imagePreview, setImagePreview] = useState(() => ({}))
+    const [dishState, setDish] = useState(() => ({
+        description: {
+        hebrew: "",
+        english: "",
+        },
+        name: {
+        hebrew: "",
+        english: "",
+        },
+        image: {},
+        isHidden: true,
+        price: 0
+    }));
+
+    function handleSubmit(e) {
+        e.preventDefault();
+        let form = new FormData();
+        _.forEach(dishState, function (value, key) {
+          if (typeof value === 'object' && key !== 'image') {
+            value = JSON.stringify(value)
+          }
+          form.append(key, value)
+        });
+        apiUtil.post('/dish', form, (res) => {
+          
+          setImagePreview(res.image.fileLink);
+          console.log(res)
+        }, (err) => console.log(err))
+    }
+
+    function update(field) {
+        if (field === "hebrew" || field === "english") {
+            return (e) => {
+              e.persist();
+                
+                setDish(oldState => {
+                  
+                    oldState[e.target.id][field] = e.target.value;
+                    let newState = Object.assign({}, oldState)
+                    return newState;
+                })
+            }
+        } else if (field === "image") {
+            return (e) => {
+                
+              e.persist()
+              setImagePreview(URL.createObjectURL(e.target.files[0]));
+              setDish((oldState) => ({
+                ...oldState,
+                [field]: e.target.files[0],
+              }));
+            };
+        } else {
+            return (e) => {
+                e.persist()
+              if (e.target.type === "checkbox") {
+                setDish((oldState) => ({
+                ...oldState,
+                [field]: e.target.checked,
+                    }));
+              } else {
+                setDish((oldState) => ({
+                    ...oldState,
+                    [field]: e.target.value
+                    }));
+                };
+            }
+        }
+    }
+    
+  return (
+    <>
+      <form onSubmit={handleSubmit}>
```

Comment:
This whole thing should be removed since you have your Form component, right?

Reply:
That's weird I deleted that file